File: moxy/compiler/src/main/java/com/omegar/mvp/compiler/entity/TagProviderMethod.java
package com.omegar.mvp.compiler.entity;

import com.omegar.mvp.presenter.PresenterType;

import javax.lang.model.type.TypeMirror;

public class TagProviderMethod {
	private final TypeMirror presenterClass;
	private final String methodName;
	private final PresenterType type;

	public TagProviderMethod(TypeMirror presenterClass, String methodName, String type, String presenterId) {
		this.presenterClass = presenterClass;
		this.methodName = methodName;
		if (type == null) {
			this.type = PresenterType.LOCAL;
		} else {
			this.type = PresenterType.valueOf(type);
		}
	}

	public TypeMirror getPresenterClass() {
		return presenterClass;
	}

	public String getMethodName() {
		return methodName;
	}

	public PresenterType getType() {
		return type;
	}

}

File: moxy/compiler/src/main/java/com/omegar/mvp/compiler/entity/TagProviderMethod.kt
package com.omegar.mvp.compiler.entity

import com.omegar.mvp.compiler.extenions.orValueOf
import com.omegar.mvp.presenter.PresenterType
import javax.lang.model.type.TypeMirror

class TagProviderMethod(
    val presenterClass: TypeMirror?,
    val methodName: String,
    val type: PresenterType,
    val presenterId: String?
) {

    constructor(
        presenterClass: TypeMirror?,
        methodName: String,
        type: String?,
        presenterId: String?
    ) : this(
        presenterClass,
        methodName,
        PresenterType.LOCAL.orValueOf(type),
        presenterId
    )

}